application/src/main/java/org/thingsboard/server/service/security/permission/Resource.java

```diff
@@ -43,25 +45,27 @@ public enum Resource {
     EDGE(EntityType.EDGE),
     RPC(EntityType.RPC),
     QUEUE(EntityType.QUEUE),
-    VERSION_CONTROL;
+    VERSION_CONTROL,
+    NOTIFICATION(EntityType.NOTIFICATION_TARGET, EntityType.NOTIFICATION_TEMPLATE,
+            EntityType.NOTIFICATION_REQUEST, EntityType.NOTIFICATION_RULE);
 
-    private final EntityType entityType;
+    private final Set<EntityType> entityTypes;
 
     Resource() {
-        this.entityType = null;
+        this.entityTypes = Collections.emptySet();
     }
 
-    Resource(EntityType entityType) {
-        this.entityType = entityType;
+    Resource(EntityType... entityTypes) {
+        this.entityTypes = Set.of(entityTypes);
     }
 
-    public Optional<EntityType> getEntityType() {
-        return Optional.ofNullable(entityType);
+    public Set<EntityType> getEntityTypes() {
+        return entityTypes;
     }
 
     public static Resource of(EntityType entityType) {
         for (Resource resource : Resource.values()) {
-            if (resource.getEntityType().orElse(null) == entityType) {
+            if (resource.getEntityTypes().contains(entityType)) {
```

Comment:
For the purpose of backward compatibility, we need to check that size of the set is 1.

Reply:
Why? Those resources that didn't have entityType will have entityTypes as empty set.
For NOTIFICATION resource the number of entity types is 4.